pallene/parser.lua

```diff
@@ -14,17 +14,28 @@ local Parser = util.Class()
 
 function Parser:init(lexer)
     self.lexer = lexer
-    self.loop_depth = 0
     self.next = false -- Token
     self.look = false -- Token
+    self.comment_regions = {} -- Sequence of sequence
```

Comment:
I've been thinking... What do you think about computing the list of comment regions in the lexer instead of in the parser? It would avoid adding extra complexity to the parser.